x-pack/plugins/reporting/public/shared/get_shared_components.tsx

```diff
@@ -40,75 +32,39 @@ export interface ApplicationProps {
   onClose: () => void;
 }
 
-/**
- * React components used to display share menus with Reporting features in an application.
- */
-export interface ReportingPublicComponents {
-  /**
-   * An element to display a form to export the page as PDF
-   * @deprecated
-   */
-  ReportingPanelPDF(props: ApplicationProps): JSX.Element;
-
-  /**
-   * An element to display a form to export the page as PDF
-   */
-  ReportingPanelPDFV2(props: ApplicationProps): JSX.Element;
-
-  /**
-   * An element to display a form to export the page as PNG
-   */
-  ReportingPanelPNGV2(props: ApplicationProps): JSX.Element;
-}
-
 /**
  * As of 7.14, the only shared component is a PDF report that is suited for Canvas integration.
  * This is not planned to expand, as work is to be done on moving the export-type implementations out of Reporting
  * Related Discuss issue: https://github.com/elastic/kibana/issues/101422
  */
-export function getSharedComponents(
+export const getSharedComponents = async (
   core: CoreSetup,
-  apiClient: ReportingAPIClient
-): ReportingPublicComponents {
-  return {
-    ReportingPanelPDF(props: ApplicationProps) {
-      return (
-        <ScreenCapturePanelContent
-          requiresSavedState={false}
-          reportType={PDF_REPORT_TYPE}
-          apiClient={apiClient}
-          toasts={core.notifications.toasts}
-          uiSettings={core.uiSettings}
-          theme={core.theme}
-          {...props}
-        />
-      );
-    },
-    ReportingPanelPDFV2(props: ApplicationProps) {
-      return (
-        <ScreenCapturePanelContent
-          requiresSavedState={false}
-          reportType={PDF_REPORT_TYPE_V2}
-          apiClient={apiClient}
-          toasts={core.notifications.toasts}
-          uiSettings={core.uiSettings}
-          theme={core.theme}
-          {...props}
-        />
-      );
-    },
-    ReportingPanelPNGV2(props: ApplicationProps) {
-      return (
-        <ScreenCapturePanelContent
-          requiresSavedState={false}
-          reportType={PNG_REPORT_TYPE_V2}
-          apiClient={apiClient}
-          toasts={core.notifications.toasts}
-          uiSettings={core.uiSettings}
-          theme={core.theme}
-          {...props}
-        />
-      );
-    },
-  };
-}
+  apiClient: ReportingAPIClient,
+  share: SharePluginSetup
+) => {
+  const [plugins, startDeps] = await core.getStartServices();
+  const { jobProviderOptions } = reportingScreenshotShareProvider({
```

Comment:
This is broken since jobProviderOptions doesn't exist, and it breaks the PDF reporting in Canvas, and the Reporting Example app. Those places should get the new modals as well.